linera-core/src/client/mod.rs

```diff
@@ -2261,73 +2162,300 @@ impl<Env: Environment> ChainClient<Env> {
-    /// Returns true if there are more certificates to fetch.
-    #[instrument(level = "trace")]
-    pub async fn sync_received_certificates_batch(&self) -> Result<bool, ChainClientError> {
+    pub async fn find_received_certificates(
+        &self,
+        cancellation_token: Option<CancellationToken>,
+    ) -> Result<(), ChainClientError> {
+        debug!(chain_id = %self.chain_id, "starting find_received_certificates");
         #[cfg(with_metrics)]
         let _latency = metrics::FIND_RECEIVED_CERTIFICATES_LATENCY.measure_latency();
-
         // Use network information from the local chain.
         let chain_id = self.chain_id;
         let (_, committee) = self.admin_committee().await?;
         let nodes = self.client.make_nodes(&committee)?;
 
-        // Proceed to downloading received certificates from a quorum of validators.
+        let trackers = self
+            .client
+            .local_node
+            .chain_state_view(chain_id)
+            .await?
+            .received_certificate_trackers
+            .get()
+            .clone();
+
+        trace!("find_received_certificates: read trackers");
+
+        let received_log_batches = Arc::new(std::sync::Mutex::new(Vec::new()));
+        // Proceed to downloading received logs.
         let result = communicate_with_quorum(
             &nodes,
             &committee,
             |_| (),
             |remote_node| {
                 let client = &self.client;
+                let trackers_clone = trackers.clone();
+                let received_log_batches = Arc::clone(&received_log_batches);
                 Box::pin(async move {
-                    client
-                        .synchronize_received_certificates_batch_from_validator(
-                            chain_id,
-                            &remote_node,
-                        )
-                        .await
+                    let batch = client
+                        .get_received_log_from_validator(&trackers_clone, chain_id, &remote_node)
+                        .await?;
+                    let mut batches = received_log_batches.lock().unwrap();
+                    batches.push((remote_node.public_key, batch));
+                    Ok(())
                 })
             },
             self.options.grace_period,
         )
         .await;
 
-        let ((), batches_with_progress) = match result {
-            Ok(result) => result,
-            Err(CommunicationError::Trusted(NodeError::InactiveChain(id))) if id == chain_id => {
-                // The chain is visibly not active (yet or any more) so there is no need
-                // to synchronize received certificates.
-                return Ok(false);
+        trace!(
+            received_log_batches_len = %received_log_batches.lock().unwrap().len(),
+            "got result from communicate_with_quorum"
+        );
+
+        if let Err(e) = result {
+            warn!(
```

Comment:
That one could actually be error-level, could it? We can still continue processing what we've got, but it makes sense to log an error if we can't reach a quorum.